lib/tarfiles.py

```diff
@@ -82,29 +88,30 @@ def do_tarballs(args):
         if tfn.startswith("dropbox:"):
             # move it to dropbox area, pretend they gave us plain path
 
-            if args.use_dropbox == "cvmfs" or args.use_dropbox == None:
+            if args.use_dropbox == "cvmfs" or args.use_dropbox is None:
                 digest, tf = slurp_file(tfn[8:])
                 proxy, token = get_creds()
 
-                if token:
-                    f = open(token,'r')
-                    tokenstr = f.read()
-                    f.close()
-
                 if not args.group:
                     raise ValueError("No --group specified!")
-                cid = "".join((args.group, "%2F", digest))
-                location = pubapi_exists(cid, proxy) # tokenstr=tokenstr
-                if not location:
-                    pubapi_publish(cid, tf, proxy) # tokenstr=tokenstr
-                    for i in range(20):
-                        time.sleep(30)
-                        location = pubapi_exists(cid, proxy) # tokenstr=tokenstr
-                        if location:
+
+                # cid looks something like dune/bf6a15b4238b72f82...(long hash)
+                cid = f"{args.group}/{digest}"
+                if args.verbose:
+                    print(f"Using RCDS to publish tarball\ncid: {cid}")
+
+                publisher = TarfilePublisherHandler(cid, proxy, token)
+                location = publisher.cid_exists()
+                if location is None:
+                    publisher.publish(tf)
+                    for i in range(NUM_RETRIES_UPLOAD):
+                        time.sleep(RETRY_INTERVAL_SEC)
```

Comment:
With the default retry values Jobsub could be sitting there for up to 10 minutes waiting for a file to become available, with no feedback to the user. Can we add a message?

Also, what happens if we hit the end of the retries and the file is still not available?

Reply:
We don't want to do that.  Note to self:  Make sure we don't wait for cid_exists to return True before moving on if we JUST published the tarball.

Reply:
Talked to Kevin - we actually will keep the logic here for now, and open a new issue for the following change:

Need to have the wrapper script do tarball discovery using the token, much like the current wrapper does.  The current way, submission will have to wait for a tarball to publish to complete.

Reply:
Will be resolved in #38.